### PR TITLE
Document one gotcha, make node content definition (via node and content fns) more flexible

### PR DESCRIPTION
**Pull request by da-z**
Hi Anthony,

You may find these changes useful too.
Regards!

### CODE REVIEW

File: docs/guide.md [hunk 1]
```diff
@@ -240,6 +240,11 @@ that didn't involve parsing their HTML if they already had it stored (like
 in a database), but also required thinking things through first. `unescaped`
 simply wraps the string in a `RawHTML` type that tells hickory to not escape it.
 
+**One gotcha here: if your node is of type `:script`, you should not use
+`unescaped` for it's `:content`.** So, if you are seeing `hickory.core.RawHTML` in your
+generated HTML, then it's probably because you're trying to escape the `:content`
+of a `:script` node.
+
```
Comment (Raynes) [inline docs/guide.md:247]: There are actually a couple of different tags that have this behavior. It should probably be handled in hickory. If you want to open an issue about it at [my hickory fork](https://github.com/Raynes/hickory), that'd be great.

File: src/me/raynes/laser.clj [hunk 2]
```diff
@@ -210,10 +213,8 @@
   "Set content of node to the string s. s can either be a collection of nodes
    or a node (a map or a string). Any strings will automatically be escaped
    unless wrapped in the RawHTML type by calling `unescaped` on them."
-  [s]
-  (fn [node] (assoc node :content (if (sequential? s)
-                                    s
-                                    [s]))))
+  [& s]
+  (fn [node] (assoc node :content (filter identity (flatten s)))))
```
Comment (Raynes) [inline src/me/raynes/laser.clj]: Why should this be desired behavior? I don't think content should take more than one argument.

File: src/me/raynes/laser.clj [hunk 3]
```diff
@@ -70,9 +72,10 @@
           :or {type :element}}]
   {:tag tag
    :type type
-   :content (if (clj/or (sequential? content) (nil? content))
-              content
-              [content])
+   :content (cond
+              (nil? content) nil
+              (sequential? content) (flatten content)
+              :else [content])
```
Comment (Raynes) [inline src/me/raynes/laser.clj:78]: I don't get the point of this one either. What is the use case for this behavior? In most cases, calling code should be handling giving these functions things they can work with, the functions shouldn't work to make weird input suitable for consumption.

File: src/me/raynes/laser.clj (hunk 2 above)
Comment (da-z) [inline src/me/raynes/laser.clj]: Is a convenience thing. One is not required now to put everything in a single vector.

Comment (Raynes) [inline src/me/raynes/laser.clj]: How is it not true? You can pass a collection of nodes. A vector, list, set, etc.

File: docs/guide.md (hunk 1 above)
Comment (da-z) [inline docs/guide.md:247]: Sure, but "Issues" is not enabled in your fork.

Comment (Raynes) [inline docs/guide.md:247]: Fixed.

**Comment by da-z** [comment]
Hi. Here is my concrete case, as I think it will be more explanatory:

``` clojure
(defn account-sign-in [email]
  (llayout {:id "sign-in"
            :masthead (static-template "plain_masthead.fragment")
            :append [(l/node :script :content (static-escaped-template "/js/focus_empty_form_fields.js"))
                     (l/node :script :content (static-escaped-template "/js/some_other_js.js"))]}
    (sign-in-page (or email ""))))
```

``` clojure
(let [document (l/parse (template "document.html"))]
  (defn llayout [opts & content]
    (l/document
      document
      (l/element= :head ) (l/content (head (:title opts) (:heads opts)))
      (l/element= :body ) (l/content
                            (when-not (:no-masthead opts) (or (:masthead opts) (static-template "masthead.fragment")))
                            (:prepend opts)
                            content
                            (:append opts)))))
```

**Comment by Raynes** [comment]
Okay, so the point is that you want to be able to pass seqs of nodes and nodes to `content` and have them all merged into one content collection? Basically, you want to pass args like

``` clojure
{:some :node}
[{:some :more} {:more :nodes}]
{:even :more})
```

and have content turn it into

``` clojure
[{:some :node} {:some :more} {:more :nodes} {:even :more}]
```

Am I correct?

**Comment by da-z** [comment]
Correct. And my scenario above works very well with these small changes I've made.

**Comment by da-z** [comment]
The reason is I am currently converting my templating code from Hiccup, where I could nest nodes (content). So not being able to do that was counter-intuitive for me.

File: src/me/raynes/laser.clj (hunk 2 above)
Comment (Raynes) [inline src/me/raynes/laser.clj]: Okay, I think I agree with you at this point that this is a good change. One last question for this one: why do you filter identity? It's perfectly fine to have `nil` in the structure, hickory will just ignore it. Was there some other reason you had to do this?

Comment (da-z) [inline src/me/raynes/laser.clj]: If it's filtered downstream than we can take that filtering out. No other reason.

Comment (Raynes) [inline src/me/raynes/laser.clj]: Cool. Remove that and I think we're good to merge this in.

Comment (da-z) [inline src/me/raynes/laser.clj]: Done

**Comment by Raynes** [comment]
I merged this in with my last string of commits. It's in the latest version, 1.1.1. Thanks for your contributions and cooperation with my incessant interrogating!

**Comment by da-z** [comment]
Always a pleasure to be interrogated :)

**Pull request closed without merging**